mantle/kola/harness.go

```diff
@@ -1222,22 +1224,15 @@ func runTest(h *harness.H, t *register.Test, pltfrm string, flight platform.Flig
 			AdditionalDisks: t.AdditionalDisks,
 			MinMemory:       t.MinMemory,
 		}
-		ioCompleted := make(chan bool)
-		go func() {
-			_, err = platform.NewMachines(c, userdata, t.ClusterSize, options)
-			ioCompleted <- true
-		}()
 
-		select {
-		case <-h.TimeoutContext.Done():
-			h.FailNow()
-		case <-ioCompleted:
-			// Finish the test
-		}
+		_, err = platform.NewMachines(c, userdata, t.ClusterSize, options)
 		if err != nil {
 			h.Fatalf("Cluster failed starting machines: %v", err)
 		}
 	}
+	// In the case of early return h.StopExecTimer can be called even
+	// if h.StartExecTimer has not been called
+	h.StartExecTimer()
```

Comment:
optional, a little more detail:
```suggestion
	// Machines should now be up. Let's start the test execution timer.
	// Each test is run via RunWithExecTimeoutCheck() in SSH() which
	// will interrupt this function/goroutine if the timeout expires.
	// In the case of early return h.StopExecTimer can be called even
	// if h.StartExecTimer has not been called
	h.StartExecTimer()
```

Reply:
as always, feel free to fix up any inaccuracies I may have included.